File: two-sum/two-sum.go
func twoSum(nums []int, target int) []int {
    seen:=make(map[int]int)
    
    for idx,val :=range nums{
        if seen_idx,ok:=seen[target-val];ok{
            return [] int {seen_idx,idx}
        }
        seen[val]=idx
    }
    
    return []int{0, 0}
}

File: two-sum/two-sum.py
class Solution(object):
    def twoSum(self, nums, target):
        """
        :type nums: List[int]
        :type target: int
        :rtype: List[int]
        """
   
        dic={}
        
        for idx,val in enumerate(nums):
            if val not in dic:
                dic[target-val]=idx
            else:
                return [idx,dic[val]]
                